javascript/code-challenges/challenges-01.test.js

```diff
@@ -0,0 +1,241 @@
+  words.forEach((word) => {
+    newWords.push(callback(word));
+  });
+  return newWords;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named addValues that takes in an array and a value and pushes the value into the array. This function does not need a return statement.
+
+Then, write a function named addNumbers that takes in four arguments:
+  - A number to be added to an array
+  - An array into which the number should be added
+  - The number of times the number should be added
+  - A callback function to use to add the numbers to the array (Hint: you already defined it)
+
+Within the addNumbers function, invoke the callback function as many times as necessary, based on the third argument of the addNumbers function.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr, value) => arr.push(value);
+
+const addNumbers = (num, arr, times, callback) => {
+  let i = 0;
+  while (i !== times) {
+    callback(arr, num);
+    i++;
+  }
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+
+CHALLENGE 6
+
+Write a function named createList that takes in an array of the current store intentory.
+
+The inventory is formatted like this:
+[
+  { name: 'apples', available: true },
+  { name: 'pears', available: true },
+  { name: 'oranges', available: false },
+  { name: 'bananas', available: true },
+  { name: 'blueberries', available: false }
+]
+
+This function should use forEach to populate your grocery list based on the store's inventory. If the item is available, add it to your list. Return the final list.
+------------------------------------------------------------------------------------------------ */
+
+const createList = (availableItems) => {
+  const list = [];
+  availableItems.forEach((item) => {
+    if (item.available) {
+      list.push(item.name);
+    }
+  });
+  return list;
+};
+
+/* ------------------------------------------------------------------------------------------------
+STRETCH - CHALLENGE 7
+
+Write a function named fizzbuzz that takes in an array of numbers.
+
+Iterate over the array using forEach to determine the output based on several rules:
+  - If a number is divisible by 3, add the word "Fizz" to the output array.
+  - If the number is divisible by 5, add the word "Buzz" to the output array.
+  - If the number is divisible by both 3 and 5, add the phrase "Fizz Buzz" to the output array.
+  - Otherwise, add the number to the output array.
+
+Return the resulting output array.
+------------------------------------------------------------------------------------------------ */
+
+const fizzbuzz = (arr) => {
+  const newArr = [];
+  arr.forEach((num) => {
+    let newNum;
+    if (num % 3 === 0 && num % 5 === 0) {
+      newNum = 'Fizz Buzz';
```

Comment:
Great work all throughout the challenges!

Something for thought here regarding variable names: What does `newNum` hold as a value? A number or a string?  If it holds a string, perhaps a different name might work here.

Other food for thought here is, is `newNum` even needed? What if you just pushed the strings directly into the array? Many ways to approach this so nice work! 👍

Reply:
I was thinking about that possibility but really wanted to get through the assignment as fast as possible

Reply:
I totally understand, there is a lot to get through.

 I will be commenting on PR's throughout the course regarding syntax and structure etc. Never feel pressured to change your code but rather take into consideration how you might approach the problem differently at another time in the future :)